pmlc/util/layout.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020, Intel Corporation
+
+#pragma once
+
+// Copyright 2020 Intel Corporation
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/Operation.h"
+#include "pmlc/util/tags.h"
+
+namespace pmlc {
+
+enum class MLFramework {
+  Default, // No changes to layouts formats will be done
+  NGraph
+};
+
+MLFramework getMLFramework(llvm::StringRef opName);
+
+enum class TensorLayout { NXC, NCX, KCX };
```

Comment:
We should use enum attributes: https://mlir.llvm.org/docs/OpDefinitions/#enum-attributes, and probably string enum attributes so that the IR is more readable (definitely worth the minor cost).

Reply:
Will move theses attributes to enum.td. The IR will be stringified.